TikTakToe. Сделал реструктуризацию решения.

m1klll opened the pull request.

m1klll committed:
Сделал реорганизацию папок

Dezodemius commented on TikTacToe/TikTakToe-pullrequest/Program.cs:
```diff
@@ -0,0 +1,206 @@
+﻿namespace TikTacToe;
```
А что ты после слияния будешь делать с этим файлом? У тебя же уже есть папка TicTacToe, а будет ещё и TicTacToe-pullrequest

Dezodemius commented on TikTacToe/TikTakToe-pullrequest/Program.cs:
```diff
@@ -0,0 +1,206 @@
+{
+    static void Main(string[] args)
+    {
+        while (true)
```
Давай перепишем цикл на тот, который имеет меньше шансов зациклиться. Например, введём какой-то флаг, и будем в цикле ориентироваться на его значение

Dezodemius commented on TikTacToe/TikTakToe-pullrequest/Program.cs:
```diff
@@ -0,0 +1,206 @@
+
+class Program
+{
+    static void Main(string[] args)
```
Давай вынесем всё в отдельный класс, а здесь будем только обращаться к нему для старта игры

Dezodemius commented on TikTacToe/TikTakToe-pullrequest/Program.cs:
```diff
@@ -0,0 +1,206 @@
+
+    static void PrintGameBoard(string[,] gameBoard)
+    {
+        Console.WriteLine("-------------");
```
"-------------" давай вынесем в отдельную константу, так как она повторяется

Dezodemius commented on TikTacToe/TikTakToe-pullrequest/Program.cs:
```diff
@@ -0,0 +1,206 @@
+        
+            if (player == "X")
+            {
+                gameBoard[row, column] = "x";
```
а почему с маленькой буквы?

Dezodemius replied:
Предлагаю сделать с большой, а литерал засунуть в константу как и "O", и везде переиспользовать

Dezodemius commented on TikTacToe/TikTakToe-pullrequest/Program.cs:
```diff
@@ -0,0 +1,206 @@
+        int stepCountPlayerOne = 0;
+        int stepCountPlayerTwo = 0;
+        
+        while (true)
```
Здесь тоже давай выберем более надёжный вариант цикла

m1klll committed:
Правки по ревью

Вынес игру в отдельный класс Game
Создал константы для обозначения полей, игроков.
Изменил логику циклов, сделал шанс зацикливания меньше
Сделал реструктуризацию
